### DISCUSSION TITLE
Listen to filter remove from filter indicators bar

### DISCUSSION THREAD

**Discussion by dreftas** [filamentphp/filament#11211] [Help] [answered]
### Package

Table builder

### Package Version

v.3.2.23

### How can we help you?

Is it possible to listen event for click on active filter removal from filters indicator tab and "reset filters" click? 

![image](https://github.com/filamentphp/filament/assets/28758219/17d8d43a-fb5e-4729-9693-952e9cbfc192)

**Comment by leandrocfe** [comment, marked as answer]
Add this method in the ListPage

```php
public function removeTableFilter(string $filterName, ?string $field = null, bool $shouldTriggerUpdatedFiltersHook = true): void
{
    parent::removeTableFilter($filterName, $field, $shouldTriggerUpdatedFiltersHook);

    // do something..
}
```

**Comment by ThisIsMyFavouriteJar** [comment]
This works great when extending a Resource from ListPage, but I get an error when trying this from a custom Page.

`class Journey extends Page implements HasTable` 

results in: _Method App\Filament\Catalog\Pages\Journey::removeTableFilter does not exist._

It seems that the removeTableFilter function is already a function in the self::class which cases an infinite loop if we use self::removeTableFilter or an error when calling parent::removeTableFilter.

Any suggestions other than copying the code from removeTableFilter directly?
